src/components/Action/Action.vue

```diff
@@ -1,3 +1,89 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import Icon from '../Icon/Icon.vue';
+import { VARIANTS, SIZES, TYPES } from './constants';
+
+defineEmits(['click']);
+const props = defineProps({
+  /**
+   * The size of the button. Defaults to big.
+   * @values big, small
```

Comment:
The correct way to do this with TS is define `type: String as PropType<Sizes>`, where `Sizes` is an enum. in typescript, it ignores JSDoc/TSDoc comments like this AFAIK. Probably in many other places.

Reply:
cool - Done